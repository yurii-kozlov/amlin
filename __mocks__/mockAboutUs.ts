/* eslint-disable max-len */
import { Feature } from 'types/aboutUs/Feature';

export const mockAboutUsData: Feature[] = [
  {
    'img': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1613170386/about/Rectangle_54_mlxq4j.png',
    'title': 'A Family That Keeps On Growing',
    'text': 'We always aim to please the home market, supplying great computers and hardware at great prices to non-corporate customers, through our large Melbourne CBD showroom and our online store.\n\nShop management approach fosters a strong customer service focus in our staff. We prefer to cultivate long-term client relationships rather than achieve quick sales, demonstrated in the measure of our long-term success.',
    'color': 'Dark',
    'icon': 'none'
  },
  {
    'img': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1613170385/about/Rectangle_11_laekoj.png',
    'title': 'shop.com',
    'text': 'Shop is a proudly Australian owned, Melbourne based supplier of I.T. goods and services, operating since 1991. Our client base encompasses individuals, small business, corporate and government organisations. We provide complete business IT solutions, centred on high quality hardware and exceptional customer service.',
    'color': 'Light',
    'icon': 'star'
  },
  {
    'img': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1613172186/about/Mask_Group_1_csjavi.png',
    'title': "You're In Safe Hands",
    'text': 'Experience a 40% boost in computing from last generation. MSI Desktop equips the 10th Gen. Intel® Core™ i7 processor with the upmost computing power to bring you an unparalleled gaming experience.\n\n*Performance compared to i7-9700. Specs varies by model.',
    'color': 'Dark',
    'icon': 'heart'
  }
]
