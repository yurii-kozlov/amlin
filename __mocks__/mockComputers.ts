/* eslint-disable max-len */
import { Product, Products } from 'types/main/Products'

export const mockComputers: Product[] = [
  {
    'inStock': true,
    'slug': 'X65v12',
    'firm': 'Artline',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528866/new-products/9_uhvi0t.jpg',
    'name': 'Computer  ARTLINE Gaming X65 (X65v12)',
    'reviewsCount': 32,
    'previousPrice': 23933,
    'price': 17856,
    'rating': 3.3,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': true,
    'slug': 'X35v17',
    'firm': 'Artline',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528866/new-products/10_wvgebo.jpg',
    'name': 'Computer  ARTLINE Gaming X35 (X35v17)',
    'reviewsCount': 3,
    'previousPrice': 15333,
    'price': 13229,
    'rating': 3.9,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': false,
    'slug': 'X39v45',
    'firm': 'Artline',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528866/new-products/12_eqpvhu.jpg',
    'name': 'Computer  ARTLINE Gaming X39 (X39v45)',
    'reviewsCount': 223,
    'previousPrice': 22933,
    'price': 19499,
    'rating': 2.5,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
]

export const mockComputersData: Products = {
  list: mockComputers,
  id: 'computers',
  firms: [
    {
      name: 'Hunter',
      id: 9
    },
    {
      name: 'Razen',
      id: 10
    },
  ]
}
