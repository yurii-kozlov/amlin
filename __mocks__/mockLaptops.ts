/* eslint-disable max-len */
import { Product, Products } from 'types/main/Products'

export const mockLaptops: Product[] = [
  {
    'inStock': true,
    'firm': 'Mi',
    'slug': 'mi-lite-15',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg',
    'name': 'Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)',
    'reviewsCount': 10,
    'previousPrice': 29999,
    'price': 21499,
    'rating': 3,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': true,
    'firm': 'Mi',
    'slug': 'redmi-13',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/3_m1wlqn.jpg',
    'name': 'Laptop Mi RedmiBook 13 i5/8/512/MX250/W (JYU4214CN)',
    'reviewsCount': 110,
    'previousPrice': 24999,
    'price': 21999,
    'rating': 3.3,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': true,
    'firm': 'Lenovo',
    'slug': 'idea-pad',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/4_waavpt.jpg',
    'name': 'Laptop Lenovo IdeaPad S145-15IKB (81VD007TRA)',
    'reviewsCount': 22,
    'previousPrice': 29999,
    'price': 15999,
    'rating': 4.1,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
]

export const mockLaptopsData: Products = {
  list: mockLaptops,
  id: 'laptops',
  firms: [
    {
      name: 'Hunter',
      id: 9
    },
    {
      name: 'Razen',
      id: 10
    },
  ]
}
