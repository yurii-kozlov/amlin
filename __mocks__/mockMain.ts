/* eslint-disable max-len */
import { Main } from 'types/main/Main';

export const mockMain: Main = {
  'banners': [
    {
      'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612461812/banners/big-banner-4_cxycmf.png',
      'id': 301
    },
    {
      'url': 'https://res.cloudinary.com/docve4syp/image/upload/c_scale,h_328,w_1398/v1670924341/techstore/banner2_lmghir.png',
      'id': 302
    }
  ],
  'newProducts': {
    'list': [
      {
        'inStock': true,
        'firm': 'Mi',
        'slug': 'mi-lite-15',
        'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg',
        'name': 'Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)',
        'reviewsCount': 10,
        'previousPrice': 29999,
        'price': 21499,
        'rating': 3,
        'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
      }
    ],
    'id': 'new'
  },
  'products': {
    'list': [
      {
        'inStock': true,
        'slug': '243V7QDAB',
        'firm': 'Philips',
        'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612530525/monitors/6_betpxe.jpg',
        'name': 'Monitor 23.8" Philips (243V7QDAB/00) Black',
        'description': '21.5 inches Full HD (1920 x 1080) widescreen IPS display\nAnd Radeon free sync technology. No compatibility for VESA Mount\nRefresh rate: 75 hertz - Using HDMI port\nZero-frame design | ultra-thin | 4ms response time | IPS panel\nPorts: 1 x HDMI & 1 x VGA\nAspect ratio - 16:9. Color supported - 16.7 million colors. Brightness - 250 nit\nTilt angle -5 degree to 15 degree. Horizontal viewing angle-178 degree. Vertical viewing angle-178 degree',
        'reviewsCount': 13,
        'previousPrice': 4499,
        'rating': 4.3,
        'price': 2553
      },
    ],
    'id': 'products',
    'firms': [
      {
        'name': 'Lenovo',
        'id': 1
      },
      {
        'name': 'Asus',
        'id': 2
      },
    ]

  },
  'news': [
    {
      'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612629196/news/image_29_1_zkln3f.png',
      'text': 'You might not suspect it, but the best budget gaming PCs are just as capable of tackling the best PC games as more expensive rigs. And, so long as you’re not expecting Ultra settings or 4K performance, a gaming PC that doesn’t come with an outrageous price tag should still be just as competent when it comes to the latest AAA games.\n\nYou may not get all the bells and whistles of the most expensive gaming rigs but the best budget gaming PCs are worth a look. A fantastic, immersive gaming experience really doesn’t require a pricey, maxed out desktop. There are plenty of less expensive, yet capable components available such as Intel’s Coffee Lake Refresh, AMD’s Ryzen 3nd Generation, not to mention the AMD Navi GPUs, that allow less expensive PCs the ability to provide that quality gaming experience. \n\nBasically, you’ll be able to fulfill all your gaming desires with the best budget gaming PCs. Whether you don’t care about gaming at the highest settings or are on a limited budget, these machines will do you justice. And, your bank account won’t be hurting either.',
      'date': '04.04.2021',
      'title': 'Gaming PCs',
      'slug': 'one'
    },
  ],

  'reviews': [
    {
      'slug': 'one',
      'name': '- Tama Brown',
      'text': 'My first order arrived today in perfect condition.  From the time I sent a question about the item to making the purchase, to the shipping and now the delivery, your company, Tecs, has stayed in touch.  Such great service.  I look forward to shopping on your site in the future and would highly recommend it.'
    },
    {
      'slug': 'two',
      'name': '- Ashot Oganisyan',
      'text': 'If your piggybank has $2,000 or more, you can choose almost any combination of features you want. Even the most powerful laptop that money can buy, though, must still obey the laws of physics. Powerful hardware generates heat, and the cooling mechanisms that such components require take up space.'
    },
  ],
  'time': '9:00 AM - 5:30 PM',
  'address': 'Kyiv, Nahimova street, 147',
  'phone': '380957737777',
  'social': {
    'fb': 'www.facebook.com',
    'instagram': 'https://www.instagram.com'
  }
}
