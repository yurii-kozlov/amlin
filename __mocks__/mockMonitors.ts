/* eslint-disable max-len */
import { Product, Products } from 'types/main/Products'

export const mockMonitors: Product[] = [
  {
    'inStock': false,
    'slug': 'mi34',
    'firm': 'Mi',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/13_qzykzl.jpg',
    'name': 'Monitor Xiaomi Mi Display 34',
    'reviewsCount': 74,
    'previousPrice': 14933,
    'price': 12499,
    'rating': 4.7,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': true,
    'slug': 'LS24R350FHIXCI2',
    'firm': 'Samsung',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/14_dhc9d6.jpg',
    'name': 'Monitor 23.8 "Samsung S24R350F (LS24R350FHIXCI) Black',
    'reviewsCount': 11,
    'previousPrice': 5933,
    'price': 4499,
    'rating': 2.8,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
  {
    'inStock': true,
    'slug': '25UM58',
    'firm': 'LG',
    'url': 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/15_kjcc9d.jpg',
    'name': 'Monitor 25 "LG 25UM58-P Black',
    'reviewsCount': 3,
    'previousPrice': 6933,
    'price': 4499,
    'rating': 5,
    'description': '15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio\nPowerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)\nAMD Radeon Vega 8 discrete graphics with Windows 10 Home\n8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD\nErgonomic backlit keyboard with fingerprint sensor activated via Windows Hello\nExclusive ErgoLift design for improved typing position'
  },
]

export const mockMonitorsData: Products = {
  list: mockMonitors,
  id: 'monitors',
  firms: [
    {
      name: 'Hunter',
      id: 9
    },
    {
      name: 'Razen',
      id: 10
    },
  ]
}
