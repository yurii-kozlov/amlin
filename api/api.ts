import axios, {AxiosResponse} from 'axios';

function get<T>(url:string):Promise<T> {
  const fullURL = process.env.NEXT_PUBLIC_BASE_URL+ url;

  return axios.get(fullURL);
}

export const getLaptops = (currentPage: number): Promise<AxiosResponse> => (
  get<AxiosResponse>(`/laptops?_limit=10&_page=${currentPage}`)
);

export const getMonitors = (currentPage: number): Promise<AxiosResponse> => (
  get<AxiosResponse>(`/monitors?_limit=10&_page=${currentPage}`)
);

export const getComputers = (currentPage: number): Promise<AxiosResponse> => (
  get<AxiosResponse>(`/computers?_limit=10&_page=${currentPage}`)
);
