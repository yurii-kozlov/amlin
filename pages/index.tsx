import { ReactElement } from 'react';
import { Header } from 'components/Header';
import { Navbar } from 'components/Navbar';
import { Slider } from 'components/Slider';
import { Services } from 'components/Services';
import { News } from 'components/News/News';
import { Logos } from 'components/Logos/Logos';
import { Footer } from 'components/Footer/Footer';
import { Reviews } from 'components/Reviews';
import { BonusSection } from 'components/BonusSection';

export default function Home(): ReactElement {
  return (
    <>
      <Header />
      <Navbar />
      <Slider />
      <BonusSection />
      <Logos />
      <News />
      <Reviews />
      <Services />
      <Footer />
    </>
  )
}
