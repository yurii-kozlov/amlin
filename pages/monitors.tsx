import React, { ReactElement } from 'react';
import { GetStaticPropsResult } from 'next';
import { Products } from 'types/main/Products';
import { instance } from 'api/api';
import { ProductsCatalog } from 'components/ProductsCatalog';

type LaptopsProps = {
  monitors: Products
}

const Monitors: React.FC<LaptopsProps> = ({ monitors }): ReactElement => (
  <ProductsCatalog products={monitors.list} />
  )

export default Monitors;

type getStaticPropsReturnMain = {
  monitors: Products
}

export async function getStaticProps():Promise<GetStaticPropsResult<getStaticPropsReturnMain>> {
      const res = await instance.get('/monitors');
      const monitors: Products = res.data;

  return {
    props: {monitors},
  }
}
