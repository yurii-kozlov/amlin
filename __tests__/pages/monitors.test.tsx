import { getStaticProps } from 'pages/monitors';
import { mockMonitorsData } from '__mocks__/mockMonitors';
import { instance } from 'api/api';

describe('Monitors page', () => {
  beforeEach(() => jest.clearAllMocks())
  test('getStaticProps returns the correct data from api', async () => {
    jest.spyOn(instance, 'get').mockImplementation(async () => ({
      data: mockMonitorsData
    }));

    const response = await getStaticProps();

    expect(instance.get).toHaveBeenCalled();
    expect(response).toMatchSnapshot();
    expect(response).toEqual({
      props: {
        monitors: mockMonitorsData
      }
    })
  })
})
export { mockMonitorsData };

