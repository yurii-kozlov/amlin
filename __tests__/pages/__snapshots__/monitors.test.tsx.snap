// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitors page getStaticProps returns the correct data from api 1`] = `
{
  "props": {
    "monitors": {
      "firms": [
        {
          "id": 9,
          "name": "Hunter",
        },
        {
          "id": 10,
          "name": "Razen",
        },
      ],
      "id": "monitors",
      "list": [
        {
          "description": "15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio
Powerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)
AMD Radeon Vega 8 discrete graphics with Windows 10 Home
8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD
Ergonomic backlit keyboard with fingerprint sensor activated via Windows Hello
Exclusive ErgoLift design for improved typing position",
          "firm": "Mi",
          "inStock": false,
          "name": "Monitor Xiaomi Mi Display 34",
          "previousPrice": 14933,
          "price": 12499,
          "rating": 4.7,
          "reviewsCount": 74,
          "slug": "mi34",
          "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/13_qzykzl.jpg",
        },
        {
          "description": "15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio
Powerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)
AMD Radeon Vega 8 discrete graphics with Windows 10 Home
8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD
Ergonomic backlit keyboard with fingerprint sensor activated via Windows Hello
Exclusive ErgoLift design for improved typing position",
          "firm": "Samsung",
          "inStock": true,
          "name": "Monitor 23.8 "Samsung S24R350F (LS24R350FHIXCI) Black",
          "previousPrice": 5933,
          "price": 4499,
          "rating": 2.8,
          "reviewsCount": 11,
          "slug": "LS24R350FHIXCI2",
          "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/14_dhc9d6.jpg",
        },
        {
          "description": "15.6 inch FHD 4 way NanoEdge bezel display with a stunning 88% screen-to-body ratio
Powerful AMD Quad Core Ryzen 5 3500U Processor (2M Cache, up to 3.6 GHz)
AMD Radeon Vega 8 discrete graphics with Windows 10 Home
8GB DDR4 RAM and 256GB PCIe NVMe M.2 SSD
Ergonomic backlit keyboard with fingerprint sensor activated via Windows Hello
Exclusive ErgoLift design for improved typing position",
          "firm": "LG",
          "inStock": true,
          "name": "Monitor 25 "LG 25UM58-P Black",
          "previousPrice": 6933,
          "price": 4499,
          "rating": 5,
          "reviewsCount": 3,
          "slug": "25UM58",
          "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612528867/new-products/15_kjcc9d.jpg",
        },
      ],
    },
  },
}
`;
